discovery_engine/lib/src/domain/models/source.dart

```diff
@@ -82,6 +82,18 @@ class AvailableSources extends Fuzzy<AvailableSource> {
           options: FuzzyOptions(
             findAllMatches: false,
             isCaseSensitive: false,
+            keys: [
+              WeightedKey(
+                name: 'name',
+                getter: (availableSource) => availableSource.name,
+                weight: 1.0,
+              ),
+              WeightedKey(
+                name: 'domain',
+                getter: (availableSource) => availableSource.domain,
+                weight: 1.0,
+              ),
+            ],
```

Comment:
❔ just out of curiosity, how does adding this option affects the outcome?

Reply:
without the key we don't get any results (or all result if the search term is the empty string), this was an oversight on my end when i implemented the search. basically, since we use a `List<AvailableSource>` instead of a `List<String>` we have to tell the searcher how to get to the string fields of the custom class.

the weights are required to give a weighting to each field/key of the searched object in the overall score computation, i've chosen them equally (they are normalized by the lib) due to lack of better values for now. eg we could give the name of the source a higher weighting than the domain of the source, depending on blue's needs.